web_fragments/fragment.py

```diff
@@ -0,0 +1,271 @@
+            placement = self._default_placement(mimetype)
+        res = FragmentResource('text', text, mimetype, placement)
+        self._resources.append(res)
+
+    def add_resource_url(self, url, mimetype, placement=None):
+        """
+        Add a resource by URL needed by this Fragment.
+
+        Other helpers, such as :func:`add_css_url` or
+        :func:`add_javascript_url` are more convenent for those common types of
+        resource.
+
+        `url`: the URL to the resource.
+
+        Other parameters are as defined for :func:`add_resource`.
+        """
+        if not placement:
+            placement = self._default_placement(mimetype)
+        self._resources.append(FragmentResource('url', url, mimetype, placement))
+
+    def add_css(self, text):
+        """
+        Add literal CSS to the Fragment.
+        """
+        self.add_resource(text, 'text/css')
+
+    def add_css_url(self, url):
+        """
+        Add a CSS URL to the Fragment.
+        """
+        self.add_resource_url(url, 'text/css')
+
+    def add_javascript(self, text):
+        """
+        Add literal Javascript to the Fragment.
+        """
+        self.add_resource(text, 'application/javascript')
+
+    def add_javascript_url(self, url):
+        """
+        Add a Javascript URL to the Fragment.
+        """
+        self.add_resource_url(url, 'application/javascript')
+
+    def add_fragment_resources(self, fragment):
+        """
+        Add all the resources from a single fragment to my resources.
+
+        This is used to aggregate resources from another fragment that
+        should be considered part of the current fragment.
+
+        The content from the Fragment is ignored.  The caller must collect
+        together the content into this Fragment's content.
+        """
+        self._resources.extend(fragment.resources)
+
+    def add_resources(self, fragments):
+        """
+        Add all the resources from `fragments` to my resources.
+
+        This is used to aggregate resources from a sequence of fragments that
+        should be considered part of the current fragment.
+
+        The content from the Fragments is ignored.  The caller must collect
+        together the content into this Fragment's content.
+        """
+        for fragment in fragments:
+            self.add_fragment_resources(fragment)
+
+    def initialize_js(self, js_func, json_args=None):
+        """
+        Register a Javascript function to initialize the Javascript resources.
+
+        `js_func` is the name of a Javascript function defined by one of the
+        Javascript resources.  As part of setting up the browser's runtime
+        environment, the function will be invoked, passing a runtime object
+        and a DOM element.
+        """
+        self.js_init_fn = js_func
+        self.js_init_version = JS_API_VERSION
+        if json_args:
+            self.json_init_args = json_args
+
+    # Implementation methods: don't override
```

Comment:
I decided to remove this since it isn't necessarily even a problem to override these methods.